be/src/olap/txn_manager.cpp

```diff
@@ -253,11 +260,11 @@ OLAPStatus TxnManager::publish_txn(OlapMeta* meta, TPartitionId partition_id, TT
     pair<int64_t, int64_t> key(partition_id, transaction_id);
     TabletInfo tablet_info(tablet_id, schema_hash, tablet_uid);
     RowsetSharedPtr rowset_ptr = nullptr;
-    WriteLock wrlock(_get_txn_lock(transaction_id));
```

Comment:
I think you may not be able to remove this lock. 
`WriteLock wrlock(_get_txn_lock(transaction_id));`

The meaning of this lock is different from that of the `_txn_map_lock`. This lock prevents a single transaction from being modified at the same time. The `_txn_map_lock` prevents the map structure from being modified at the same time.

If this lock is removed here. Then there may be two threads publishing the txn at the same time, which will modify their contents(such as rowset meta) at the same time.